models/base/scratch/base_scratch.yml

```diff
@@ -262,11 +262,15 @@ models:
       - name: root_id
         description: '{{ doc("col_root_id") }}'
         tests:
-          - not_null
+          - not_null:
+              config:
+                enabled: '{{var("snowplow__enable_application_context")}}'
```

Comment:
So I think you will run into the [same bug](https://github.com/dbt-labs/dbt-core/issues/3698) as I did when using a similar approach with the web package. Worth double checking but I think currently if you have enabled `snowplow__enable_application_context` within your root project, the model will execute but the tests on it will be disabled still. The work around I chose was to set default values for all these vars inside the actual `var` rather than in the packages `dbt_project.yml` file. The user can then override these from their root project. That way I believe both the model and tests should be enabled/disabled together.

https://github.com/snowplow/dbt-snowplow-web/blob/c0adfb1a275bf7dc2a428345b7d87b438cf56718/dbt_project.yml#L37-L42

Reply:
I'm not sure why but they seem to be coming through for me without relegating the `var` definition to the `schema.yml` files. Here you can see what happens when I enable all of the contexts except the geo one:
![image](https://user-images.githubusercontent.com/2262032/151148147-b493f690-799b-443c-96d3-2c205ba00341.png)
and here when they are all disabled:
![image](https://user-images.githubusercontent.com/2262032/151148292-7eca769a-301b-4729-bc09-636329992f57.png)
Again using only the definition in the `dbt_project.yml`

Reply:
Which version of dbt are you using?

Reply:
Also are you importing the package into a dbt project or running directly from within the package?

Reply:
Ah yep sorry, I was running directly from within the package. I guess it's expected behaviour from dbt's perspective, right? According to this page (https://docs.getdbt.com/docs/building-a-dbt-project/building-models/using-variables#variable-precedence), the package variables have precedence over global variables. But I've now changed them to be defined throughout the project with default value of `false`, thanks for asking!

Reply:
Not quite sure what behaviour you are referring two but having the same var return different values within the package is not expected :)